src/services/redisSessionStorage.ts

```diff
@@ -729,41 +706,67 @@ export default class RedisSessionStorage extends RedisStorageUtils
         _ => [],
         _ =>
           _.map(deserializedUser => {
-            return [
-              deserializedUser.wallet_token !== user.wallet_token
-                ? some(`${walletKeyPrefix}${deserializedUser.wallet_token}`)
-                : none,
-              UserV2.is(deserializedUser) &&
-              deserializedUser.myportal_token !== user.myportal_token
-                ? some(
-                    `${myPortalTokenPrefix}${deserializedUser.myportal_token}`
+            return collect(
+              this.getUserTokens(deserializedUser)
+                // Filter keys already contained in oldSessionInfoKeys and oldSessionKeys
+                .filter(
+                  not(
+                    p =>
+                      p.prefix === sessionInfoKeyPrefix ||
+                      p.prefix === sessionKeyPrefix
+                  )
+                )
+                // Filter wallet_token of the new session
+                .filter(
+                  not(
+                    p =>
+                      p.prefix === walletKeyPrefix &&
+                      p.value === user.wallet_token
+                  )
+                )
+                // Filter myportal_token and bpd_token of the new session
+                .filter(
+                  and(
+                    not(
+                      p =>
+                        p.prefix === myPortalTokenPrefix &&
+                        p.value === user.myportal_token
+                    ),
+                    not(
+                      p =>
+                        p.prefix === bpdTokenPrefix &&
+                        p.value === user.bpd_token
+                    )
```

Comment:
is it feasible to put everything under the same` filter() `call?

maybe we can use something like `p.wallet_token.value != user.wallet_token` instead of matching the prefix using `filterWithKey`?

Reply:
I've created a new `and` function that compose more than 2 predicates in da96bdb, What do you think about?

Reply:
can we just use boolean operators instead of methods? ie:

`p =>  !(p.prefix === myPortalTokenPrefix && p.value === user.myportal_token) && ..`

Reply:
fixed in e81d09b